include/pokeathlon/pokeathlon.h

```diff
@@ -4,10 +4,38 @@
 #include "constants/heap.h"
 
 #include "overlay_manager.h"
+#include "player_data.h"
+#include "save.h"
+
+// Forward declaration
+typedef struct PokeathlonCourseData PokeathlonCourseData;
+
+// Function pointer types for state management
+typedef BOOL (*PokeathlonStateFunc)(PokeathlonCourseData *, void **);
+typedef BOOL (*PokeathlonStateHandlerFunc)(PokeathlonCourseData *, int);
+
+// State info structure (8 bytes)
+typedef struct PokeathlonStateInfo {
+    void *ptr;        // 0x00 - Pointer to state function table
+    u8 field_04;      // 0x04 - Field at offset 4 (initialized to 1)
+    u8 stateArgsBase; // 0x05 - Base for state function args
+    u8 stateIndex;    // 0x06 - State function index
+    u8 field_07;      // 0x07 - Field at offset 7
+} PokeathlonStateInfo;
+
+// Participant data structure (0x7C = 124 bytes each)
+typedef struct PokeathlonParticipantData {
+    u8 data[0x7C]; // Opaque data for now
```

Comment:
As explained in the other comment, there appears to be a function (`PokeathlonCourse_GetParticipantData2`) that returns the 4th element of this array. May be worth to un-opaque that one element (?), like

```suggestion
    u8 field_00[0x03];
    u8 field_04;
    u8 field_05[0x78];
```

Not sure if this is something we usually do, though.